### PR TITLE
v2 first draft

### PR DESCRIPTION
**Pull request by pippo314**

### CODE REVIEW

File: bin/python [hunk 1]
```diff
@@ -0,0 +1 @@
+python3
```
Comment (e-caste) [inline bin/python:1]: Is this file needed? Also `bin/python3` and `lib64`

File: pytarallo/Item.py [hunk 2]
```diff
@@ -1,60 +1,62 @@
...
-                # setattr(self, k, v)
-                self.features[k] = v
-
             # setup path and location
-            # location: the most specific postion where the item is located, e.g. "Table"
+            # location: the most specific position where the item is located, e.g. "Table"
             # path: a list representing the hierarchy of locations the item, e.g. ["Polito", "Chernobyl", "Table"]
             # the tarallo server returns an JSON object containing only the "location" attribute, which corresponds
             # to the path as defined above
+            if data.get('code') is not None:
```
Comment (e-caste) [inline pytarallo/Item.py:31]: Smart use of `get` to not  write he more verbose `'code' in data and data['code']`.
In Python, `None` is falsy, so we can omit the check, especially if code is a string (don't do this if code can be an empty list or other falsy variables):
```suggestion
            if data.get('code'):
```
You can apply this suggestion to other similar lines as well.

File: pytarallo/ProductToUpload.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from pytarallo.Product import Product
+
+
+class ProductToUpload(Product):
+    """Not sure what this is"""
```
Comment (e-caste) [inline pytarallo/ProductToUpload.py:5]: Uhm... can we maybe clarify this a little bit better? Calling in @lvps for extra support

File: pytarallo/Tarallo.py [hunk 4]
```diff
@@ -1,12 +1,16 @@
 import json
 import urllib.parse
-import requests
+import pytarallo.Errors as Errors
```
Comment (e-caste) [inline pytarallo/Tarallo.py:3]: This _should_ be equivalent:
```suggestion
import Errors
```

File: pytarallo/ProductToUpload.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+from pytarallo.Product import Product
```
Comment (e-caste) [inline pytarallo/ProductToUpload.py:1]: We are in the `pytarallo` directory/module already, so this should work:
```suggestion
from Product import Product
```

File: pytarallo/Tarallo.py [hunk 6]
```diff
@@ -1,12 +1,16 @@
...
 from typing import Optional
+from pytarallo.Product import Product
+
+import requests
 
+from pytarallo import ProductToUpload, ItemToUpload
 from pytarallo.AuditEntry import AuditEntry, AuditChanges
-from pytarallo.Errors import *
 from pytarallo.Item import Item
```
Comment (e-caste) [inline pytarallo/Tarallo.py:11]: Same as in `Tarallo.py`, are we not already in the `pytarallo` module? 
Maybe we need to add an empty `__init__.py` directory to tell Python this is a module if it doesn't exist already

File: pytarallo/Tarallo.py [hunk 7]
```diff
@@ -98,41 +102,92 @@ def status(self):
...
             return self.get('/v2/session').status_code
-        except AuthenticationError:
+        except Errors.AuthenticationError:
             return self.response.status_code
 
     def get_item(self, code, depth_limit=None):
-        """This method returns an Item instance"""
-        url = '/v2/items/' + self.urlencode(code)
+        """This method returns an Item instance received from the server"""
+        url = '/v2/items/' + self.urlencode(code) + '?separate' # try an Item without product6
```
Comment (e-caste) [inline pytarallo/Tarallo.py:110]: Maybe we should use fstrings everywhere to improve readability. See also following lines.

File: pyvenv.cfg [hunk 8]
```diff
@@ -0,0 +1,3 @@
+home = /usr/bin
+include-system-site-packages = false
+version = 3.6.9
```
Comment (e-caste) [inline pyvenv.cfg:3]: Please add this file to .gitignore or better yet, delete it and switch to using `venv` (it's also integrated in PyCharm).

**Review by e-caste — CHANGES_REQUESTED** [review]
Some changes requested, mostly formal. See the comments for my notes, also about some files which I don't quite understand why are required.

Overall, this looks like a nice job! But to make sure: are the tests complete? Do they pass? Have you tested them against a development instance of Tarallo?

Of course, we will implement a CI workflow to automatically run the tests when they will (or already are) complete, like in some of our other repositories.

File: pytarallo/ProductToUpload.py (hunk 3 above)
Comment (lvps) [reply pytarallo/ProductToUpload.py:5]: Well, it's a product and it should be uploaded asd
Can you make it inherit from the Product class, so it's the same thing with a different name? IIRC there's no difference right now, but there may be in the future, at least we're prepared, especially in code that will use this library (i.e. we don't have to review every use of Product to check if it should be ProductToUpload when they start to differ in some way)

File: pytarallo/Tarallo.py (hunk 7 above)
Comment (lvps) [reply pytarallo/Tarallo.py:110]: Agree, fstrings are just so much more readable than any other form of string concatenation
